### ISSUE TITLE
Enumeration failed crash (avalonia)

### ISSUE THREAD

**Issue by Jon-Murray** [beto-rodriguez/LiveCharts2#643]
Hi Beto,

   So first off, thanks for the library. It's really great and has been a pleasure to use. I have quite an odd bug which keeps triggering for me, and i'm able to reproduce it. I believe it is related to https://github.com/beto-rodriguez/LiveCharts2/issues/225 but i'm not entirely sure. I have set up a repo to outline the issue https://github.com/Jon-Murray/livecharts-issue

Right, so now an overview as to what it is and what is happening. The demo will create a load of charts in a row, wrapped inside a scrollviewer. Each chart has its own thread inside the viewmodel which will add data, making sure each chart is "unique". nb. there's no locking in there at the moment, as each thread is local to each chart, but in a "prod" scenario this is called via an "additem" method exposed from the viewmodel, which locks, adds, and returns. I've also added in the SyncContext="locker" to the graph view just in case there is something going on internally with locking and the UI thread, but this doesn't seem to make a difference.

How to trigger it? Well, this is the weird bit. I've found a way to reliably(ish) trigger it, but not fully reliably. Basically, run the example, scroll to the bottom to ensure all graphs are rendered, wait for about 60-90 seconds for the graphs to get populated with some data and then go back into the main window and start scrolling up and down really rapidly (if you're using a trackpad, think like a DJ scratching a vinyl). Eventually - sometimes it can take 20-30 seconds - it will crash (there's a global exception handler in program.cs, if you wanted to put a breakpoint there). An example error message is below. 

I believe i'm doing everything correctly with regards to locking, but i've not seen anyone else use a class in an observablecollection like i am doing in GraphVM.cs (it makes things easier in my production application), but seems to work alright like this so is likely unrelated. I am running on Windows 10. 

Thanks!

```"   at System.ThrowHelper.ThrowInvalidOperationException_InvalidOperation_EnumFailedVersion()\r\n   at System.Collections.Generic.List`1.Enumerator.MoveNext()\r\n   at LiveChartsCore.Kernel.Providers.DataFactory`2.<EnumerateByRefEntities>d__20.MoveNext()\r\n   at LiveChartsCore.Kernel.Providers.DataFactory`2.<Fetch>d__10.MoveNext()\r\n   at LiveChartsCore.Measure.MeasureExtensions.<YieldReturnUntilNextNullChartPoint>d__2.MoveNext()\r\n   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()\r\n   at LiveChartsCore.Measure.MeasureExtensions.<AsSplineData>d__1.MoveNext()\r\n   at LiveChartsCore.LineSeries`6.<GetSpline>d__27.MoveNext()\r\n   at LiveChartsCore.LineSeries`6.Measure(Chart`1 chart)\r\n   at LiveChartsCore.CartesianChart`1.Measure()\r\n   at LiveChartsCore.Chart`1.<UpdateThrottlerUnlocked>b__145_1()\r\n   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority)\r\n   at Avalonia.Win32.Win32Platform.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)\r\n   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)\r\n   at Avalonia.Win32.Win32Platform.RunLoop(CancellationToken cancellationToken)\r\n   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)\r\n   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)\r\n   at Program.Main(String[] args) in C:\\Users\\Jon\\Desktop\\Stuff\\Code\\Program.cs:line 17"```

**Comment by Jon-Murray** [comment]
This seems to be because I just had {locker} and not {Binding locker}. Sometimes the smallest bugs take the longest time. At least I know a lot more about livecharts internals now after having spent all day debugging them, how embarrassing. So sorry for time wasting. Thanks.

**Comment by beto-rodriguez** [comment]
Glad you found it!

There is nothing to be embarrassed! it needs better docs

**Comment by Jon-Murray** [comment]
Hi Beto, it seems I made this in error (was debugging and saw something obvious and forgot how my example worked). Ive narrowed it down to something to do with the two core threads (update throttled and measuring) having a race condition when auto updates are enabled with the core points array. I'm not in front of the computer at the moment, but that's as far as I got. It's likely avalonia specific. I'm going to have a proper look over it over the next few days, as ostensibly it all looks correct with locking. If you increase the frequency of updates in my example to 100ms it should trigger a crash a lot more easily.

Sorry again!

28 Sept 2022 18:35:43 Alberto Rodríguez ***@***.***>:
